nodejs/awsx/ec2/vpc.ts

```diff
@@ -49,18 +50,47 @@ export class Vpc extends pulumi.ComponentResource {
      */
     public readonly natGateways: x.ec2.NatGateway[] = [];
 
-    constructor(name: string, args?: VpcArgs, opts?: pulumi.ComponentResourceOptions);
-    constructor(name: string, args?: ExistingVpcArgs, opts?: pulumi.ComponentResourceOptions);
-    constructor(name: string, args: VpcArgs | ExistingVpcArgs = {}, opts: pulumi.ComponentResourceOptions = {}) {
-        super("awsx:x:ec2:Vpc", name, {}, opts);
+    private readonly parent: Vpc;
+
+    constructor(name: string, parent: Vpc,
+                args: InternalVpcArgs | ExistingVpcArgs | ExistingVpcIdArgs,
+                opts: pulumi.ComponentResourceOptions = {}) {
+        this.parent = parent;
 
         if (isExistingVpcArgs(args)) {
             this.vpc = args.vpc;
             this.id = this.vpc.id;
         }
+        else if (isExistingVpcIdArgs(args)) {
+            this.vpc = aws.ec2.Vpc.get(name, args.vpcId, {}, opts),
+            this.id = this.vpc.id;
```

Comment:
this code just inlines `Vpc.fromExistingIds`.  But it means we don't need constructors/facotry methods on Vpc and VpcData.  in VpcData it's all just constructors.